certauth.go

```diff
@@ -136,7 +149,7 @@ func (a *Auth) ValidateRequest(r *http.Request) error {
 }
 
 // Process is the main Entrypoint
-func (a *Auth) Process(w http.ResponseWriter, r *http.Request) error {
+func (a *Auth) Process(w http.ResponseWriter, r *http.Request, site string) error {
```

Comment:
What if the `AuthorizationChecker` can just take what it needs from the `http.Request`? Then I think we won't need to change this signature. Otherwise we should v2 the module, IIUC.

Reply:
Oh, I had kinda assume most modules did what `go-demo-service` does and don't call this `Process` method directly...

I'll do a cursory check to make sure that nothing is directly calling `Process`. If stuff is, I'll restructure this to not break things, but if there are no callsites outside this repo, I think this is fine as-is?

Reply:
Proving that there are no callsites outside the repo sounds harder than making a v2 where this *not* exported if that's our expectation. That way we can't break anyone who isn't explicitly opting in, and therefore open to changing their code.

Reply:
That's a public method in the public repo.

Reply:
> What if the AuthorizationChecker can just take what it needs from the http.Request? 

Do you mean header or context?

Reply:
> That's a public method in the public repo.

So? We're allowed to break whatever we want if we do a `v2`. If we want to keep the `Process` function exported, that's fine too, but if we want to generally be able to change it because we *don't* want people calling it, then we should stop exporting it.

> > What if the AuthorizationChecker can just take what it needs from the http.Request?
> 
> Do you mean header or context?

I mean the whole request. I could see wanting *not* to pass the whole request from a perspective of limiting the degree to which the consuming code has to trust this codebase, but just headers or just context requires something else to stuff the TLS termination information into one or both of those places.

Reply:
@winmillwill, right, I'm just confirming your words.

Reply:
> What if the AuthorizationChecker can just take what it needs from the http.Request?

The `AuthorizationChecker` doesn't have access to the `http.Request` right now. Only the site ID derived from the request's URI parameters.

I could make `Process` pull the site ID out instead of doing it in `RouterHandler` but it seems not ideal to parse the arguments twice.

Instead I'm going to restructure `Process` so it keeps it's signature and add a `ProcessWithSite` which accepts the new parameter. This will keep current call sites compiling as is with the current behavior and also allow us to parse the site-id exactly once.